plugins/cocoapods/src/index.ts

```diff
@@ -144,34 +187,86 @@ export default class CocoapodsPlugin implements IPlugin {
       auto.prefixRelease(getVersion(this.options.podspecPath))
     );
 
-    auto.hooks.version.tapPromise(this.name, async ({ bump, dryRun, quiet }) => {
-      const previousVersion = getVersion(this.options.podspecPath);
-      const releaseVersion = inc(previousVersion, bump as ReleaseType);
+    auto.hooks.version.tapPromise(
+      this.name,
+      async ({ bump, dryRun, quiet }) => {
+        const previousVersion = getVersion(this.options.podspecPath);
+        const releaseVersion = inc(previousVersion, bump as ReleaseType);
+
+        if (dryRun && releaseVersion) {
+          if (quiet) {
+            console.log(releaseVersion);
+          } else {
+            auto.logger.log.info(`Would have published: ${releaseVersion}`);
+          }
 
-      if (dryRun && releaseVersion) {
-        if (quiet) {
-          console.log(releaseVersion);
-        } else {
-          auto.logger.log.info(`Would have published: ${releaseVersion}`);
+          return;
         }
 
-        return;
-      }
+        if (!releaseVersion) {
+          throw new Error(
+            `Could not increment previous version: ${previousVersion}`
+          );
+        }
 
-      if (!releaseVersion) {
-        throw new Error(
-          `Could not increment previous version: ${previousVersion}`
-        );
+        updatePodspecVersion(this.options.podspecPath, releaseVersion);
+
+        await execPromise("git", [
+          "commit",
+          "-am",
+          `"update version: ${releaseVersion} [skip ci]"`,
+          "--no-verify",
+        ]);
+
+        await execPromise("git", [
+          "tag",
+          releaseVersion,
+          "-m",
+          `"Update version to ${releaseVersion}"`,
+        ]);
       }
+    );
 
-      await updatePodspecVersion(this.options.podspecPath, releaseVersion);
-      await execPromise("git", [
-        "tag",
-        releaseVersion,
-        "-m",
-        `"Update version to ${releaseVersion}"`,
-      ]);
-    });
+    auto.hooks.canary.tapPromise(
+      this.name,
+      async ({ bump, canaryIdentifier, dryRun, quiet }) => {
+        const pr = getPrNumberFromEnv();
+
+        if (!auto.git || !pr) {
```

Comment:
my only remaining question, since cocoapods only works with commits in git i basically NEED the PR number to get the remote url, should i just add a log statement here if they try to run it locally?

Reply:
or i can default to `auto.remote` if the PR number is not available and if they havent pushed the commit up to their origin, it will fail to deploy locally, but if theyve pushed it will work, and i can call it out in the docs

Reply:
i feel like i answered my own question, going to push up a commit for this